src/app/header/header-profile/header-profile.component.ts

```diff
@@ -30,6 +44,71 @@ export class HeaderProfileComponent implements OnInit, OnDestroy {
     });
   }
 
+  private subscribeToUser(): void {
+    this.authService.user.pipe(
+      takeUntil(this.destroy$)
+    ).subscribe(user => {
+      this.user = user;
+      this.cartService.getCount();
+    });
+  }
+
+  private subscribeToProductButtonClick(): void {
+    this.authService.onProductButtonClick.pipe(
+      takeUntil(this.destroy$)
+    ).subscribe((ifToggle) => {
```

Comment:
```suggestion
    ).subscribe((isToggled: boolean) => {
```

Reply:
sorry, it is because IDE knows about it. Should I put the type anyway?
![image](https://user-images.githubusercontent.com/64072558/102875917-217ddf80-444d-11eb-81be-5c5ba9340e87.png)

Reply:
IDE knows it, but I don't know what I expect to receive there (unless I will not find this variable). We are watching your PR not in IDE... You can skip sometimes type definition if for example you want to map through array of smth. `products.map(product => {})`, however if you have a big nesting, there type definition would be helpful

Reply:
@maksym-syniuk ok nice! it is fixed and I will consider it for future

Reply:
Great!